integration-tests/bats/global.bats

```diff
@@ -0,0 +1,380 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+}
+
+teardown() {
+    assert_feature_version
+    teardown_common
+}
+
+@test "global: basic case" {
```

Comment:
These tests are good, but I think it's probably worthwhile to write some engine test scripts as well. Better chance somebody finds a breaking change before it hits CI in that case.